packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -582,22 +587,137 @@ class CommentsRepo extends AbstractRepo<"Comments"> {
       LEFT JOIN "CommentEvents" ce ON c._id = ce."documentId"
       ORDER BY COALESCE(c."topLevelCommentId", c._id), c."postedAt"
       LIMIT $(maxTotalComments) -- Apply final limit
-    `, { userId, initialCandidateLimit, maxTotalComments });
+    `, { 
+      userId, 
+      initialCandidateLimit, 
+      maxTotalComments,
+      initialCandidateLookbackDaysParam: initialCandidateLookbackDays,
+      commentServedEventRecencyHoursParam: commentServedEventRecencyHours
+    });
 
     return feedCommentsData.map((comment): FeedCommentFromDb => ({
       commentId: comment.commentId,
+      authorId: comment.authorId,
       topLevelCommentId: comment.topLevelCommentId,
       parentCommentId: comment.parentCommentId ?? null,
       postId: comment.postId,
       baseScore: comment.baseScore,
       shortform: comment.shortform ?? null,
       postedAt: comment.postedAt,
-      sources: ['recentComments'], // Assign fixed source
+      sources: ['recentComments'],
       lastServed: null, 
       lastViewed: comment.lastViewed ?? null,
       lastInteracted: comment.lastInteracted ?? null,
     }));
   }
+
+  /**
+   * Fetches consolidated engagement statistics for recently active comment threads.
+   */
+  async getThreadEngagementStatsForRecentlyActiveThreads(
+    userId: string,
+    threadEngagementLookbackDays: number
+  ): Promise<ThreadEngagementStats[]> {
+    const db = this.getRawDb();
```

Comment:
Don't do this, please (and fix the other instances).  Our existing pattern has been to do this inline, so doing it elsewhere makes it confusing to read the query and then wonder how you're fetching non-collection-shaped rows without going through getRawDb first.